scipy/optimize/tests/test_zeros.py

```diff
@@ -169,3 +169,36 @@ def f_2(x, *a):
     y = zeros.newton(f, z, args=coeffs, fprime=f_1, fprime2=f_2, tol=1e-6)
     # (-0.75000000000000078+1.1989578808281789j)
     assert_allclose(f(y, *coeffs), 0, atol=1e-6)
+
+
+@pytest.mark.filterwarnings("error")
```

Comment:
this line can be removed, this is the default behavior (see `pytest.ini` line 7 in the repo root).

Reply:
Hi @rgommers, even though the `error` warning filter is set by default, it didn't seem to work when I use `python runtest.py -t scipy/optimize/tests/test_zeros.py` and more importantly I feel that it's more explicit to leave the decorator even if it is redundant.

Reply:
We want to control this globally. For one, we're switching this behavior between development versions and releases (we do want errors for the former, not for the latter).

Reply:
Okay